src/main.cpp

```diff
@@ -3422,6 +3440,12 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
             } else {
                 pindex->nChainTransparentValue = std::nullopt;
             }
+
+            if (pindex->pprev->nChainLockboxValue) {
+                pindex->nChainLockboxValue = *pindex->pprev->nChainLockboxValue + lockboxValueDelta;
+            } else {
+                pindex->nChainTransparentValue = std::nullopt;
+            }
```

Comment:
Bug: we are missing the equivalent logic to load `nChainLockboxValue` into memory on start in `LoadBlockIndexDB`.